Adding clip duration to Events View

banthungprong opened the pull request:
Issue #3813 
(I'm not sure if this is the correct workflow, but I try...:
In the GitHub Docs it should be possible to manually link an issue to this pull request via the Development in the sidebar. Is not possible here).

Simply added the difference between end and start time as a span-Block.
And in the next div after the Start-Time the End-Time.
![frigate Events with Length](https://user-images.githubusercontent.com/59219161/196948039-aae63cce-2c55-4fbf-8b3a-1972ae46b6e9.jpg)

banthungprong committed:
Adding clip length in s to Events View

NickM-27 commented:
Happy to hear what others think, but my thoughts are:

- having the end and start time as well as duration in seconds is redundant. 
- Maybe an option in the config could do one or the other. 
- Just doing seconds is fine for short events but when you have events for a parked car `6372s` is not super intuitive, might be nice if it was broken intona separate function and formatted as `h m s`

NickM-27 reviewed (changes requested):
Happy to hear what others think, but my thoughts are:

- having the end and start time as well as duration in seconds is redundant. 
- Maybe an option in the config could do one or the other so users have a choice 
- Just doing seconds is fine for short events but when you have events for a parked car `6372s` is not super intuitive, might be nice if it was broken into a separate function and formatted as `h m s`

banthungprong commented:
> Happy to hear what others think, but my thoughts are:
> 
> * having the end and start time as well as duration in seconds is redundant.
> * Maybe an option in the config could do one or the other so users have a choice
> * Just doing seconds is fine for short events but when you have events for a parked car `6372s` is not super intuitive, might be nice if it was broken into a separate function and formatted as `h m s`

Thanks for you fast reply!
* end/start time is redundant. Will remove it.
* will add an option in the config (because I did this already with an option to enable/disable a camera in the config (#4109). Default will be disabled
* formatting is natural (but I thought that the events never would reach these values). Will change the output.

blakeblackshear commented:
I would rather not end up with a bunch of config options that modify the UI. Let's just show the start time and the duration. That should cover the vast majority of needs.

NickM-27 commented:
Sounds good to me, @banthungprong in that case can also move the duration down to the other time fields since there won't be an end time 👍

banthungprong commented:
I formatted it like this (hours/minutes only shown if not 0), using a new function clipLength():
![image](https://user-images.githubusercontent.com/59219161/196970850-8e153bc0-7c7f-473c-be5d-046e66ec2de1.png)

banthungprong committed:
added function returning human readable length

NickM-27 commented on web/src/routes/Events.jsx:
```diff
@@ -37,6 +37,21 @@ const monthsAgo = (num) => {
+  let minutes = Math.floor(totalSeconds / 60);
+  let seconds = (totalSeconds % 60).toFixed(0);
+  let length=`${hours} Hours ${minutes} Minutes ${seconds} Seconds`
+  if (hours == 0 && minutes == 0) {
+    length=`${seconds} Seconds`
+  } else {
+    length=`${minutes} Minutes ${seconds} Seconds`
+  }
```
What if we used abbreviations instead?


```suggestion
  if (hours == 0 && minutes == 0) {
    length=`${seconds}s`
  } else {
    length=`${minutes}m ${seconds}s`
  }
```

menkej replied:
I like that but I'm not 100% sure what that means for translations... Is m and s really international (thinking about Asian languages, etc)?

banthungprong replied:
In the final commit "Minutes/Seconds" are used (identical to the existing solution in RecordingPlaylist.jsx).
For Translation see #947 (that's what I found regarding internationalization/language resources).

NickM-27 commented on web/src/routes/Events.jsx:
```diff
@@ -459,7 +474,7 @@ export default function Events({ path, ...props }) {
                         <div className="text-sm">
                           {new Date(event.start_time * 1000).toLocaleDateString()}{' '}
-                          {new Date(event.start_time * 1000).toLocaleTimeString()}
+                          {new Date(event.start_time * 1000).toLocaleTimeString()} <span className="text-red-300">({clipLength(event.end_time - event.start_time)})</span>
```
I don't think we want to color the text here, the red doesn't mean anything but it could imply an error or some other meaning which isn't important as it is just the duration.

banthungprong replied:
- red: correct. Simply took the color from the original request. Will remove it.
- Abbreviations are ok for me here. But: In the Recordings View "Minutes"/"Seconds" are used (RecordingPlaylist.jsx). I assumed it would be better to use the same formatting in different views.
- in RecordingPlaylist.jsx functions from "date-fns" are used. So I will do the same here - stupid to implement something new if you get it for free...
- When a clip is new, the end time and duration is still "in progress". So I got negative Integer as results... The solution was already in the RecordingPlaylist.jsc.

banthungprong committed:
switched to date-fns functions for formatting

banthungprong committed:
fixed switched start/end time, changed length to duration

banthungprong commented:
I found a "stupid" mistake. Switched start/end time in the function call. Found this when using a docker image based on this branch.
Renamed function to clipDuration because a lenght would be Bytes a duration is a time range.
